### ISSUE TITLE
ter-arrow-parens don't recognize the need of parens

### ISSUE THREAD

**Issue by Alecell** [buzinas/tslint-eslint-rules#391]
I have this case in my code:
```
const texts = [
  (leadInfos?) => ({
    name: 'Test',
    text: '',
  }),
  (leadInfos) => ({
    name: 'Test',
    text: '',
  }),
  (leadInfos?) => ({
    name: 'Test,
    text: '',
  }),
];
```

The `ter-arrow-parens` throws `Unexpected parentheses around single function argument having a body with no curly braces.` 3 times in sequence, but the parens is required on the first and last case, it should throw this error only on the second case since the `?` is used to define conditional parameters.
